src/99-appendices/15-inheritance-principle.md

```diff
@@ -0,0 +1,260 @@
+
+1.  If, for a given data file, multiple metadata files satisfy criteria 2.a-c above:
+
+    1.  The set of applicable metadata files is ordered as follows.
+        Within each level of the filesystem hierarchy independently,
+        applicable files are ordered from fewest to most entities;
+        each subsequent filename MUST possess a set of entities that is a
+        strict superset of the previous filename.
+        These lists are concatenated in order of highest to lowest level in the
+        filesystem hierarchy.
+
+    1.  For [tabular files](#tabular-files) and other simple metadata files
+        (for instance, [`bvec` / `bval` files for diffusion MRI](#bvec-bval),
+        accessing metadata associated with a data file MUST consider only the
+        last file in the order established by rule 4.a.
+
+    1.  For [JSON files](#json-files), key-values MUST be loaded
+        from applicable files sequentially in the order established by rule 4.a,
+        overwriting any existing key-values when doing so.
+
+### Corollaries
+
+1.  As per rule 3, metadata files applicable only to a specific participant / session
+    MUST be defined in or below the directory corresponding to that participant / session;
+    similarly, a metadata file that is applicable to multiple participants / sessions
+    MUST NOT be placed within a directory corresponding to only one such participant / session.
+
+1.  It is permissible for a single metadata file to be applicable to multiple data files.
+    Where such metadata content is consistent across multiple data files,
+    it is RECOMMENDED to store metadata in this way,
+    rather than duplicating that metadata content across multiple metadata files.
+
+1.  Where multiple applicable [JSON files](#json-files) are loaded
+    for one data file as per rules 4.a and 4.c:
+
+    1.  Where the same key is present in multiple applicable metadata files,
+        the final value associated with that key will be that of the file latest in
+        the order in which that key is defined;
+        any values associated with that key earlier in the ordering are overridden
+        (though it is RECOMMENDED to minimize the extent of such overrides).
+
+    1.  The requirement that applicable metadata files follow a strict ordering according to
+        addition of entities applies individually within each level of the filesystem hierarchy;
+        it is not necessary for the complete ordered set of all applicable metadata files
+        across all filesystem levels to demonstrate strict addition of entities between
+        sequential filenames.
+
+    1.  A key-value being present in a metadata file earlier in the ordering but absent in
+        any file later in the ordering does not imply the "unsetting" of that key-value.
+
+    1.  Removal of key-values present in files earlier in the ordering based on the content
+        of files later in the ordering is not possible.
```

Comment:
do you have any changes for this PR planned @Lestropie or may be it should be taken out of Draft mode and gain a chance to be Reviewed more closely and merged? ;-)

Reply:
Have made some proposed changes based on comment threads above.

I'm slightly unsure about pulling this out of draft given #1195. My suspicion is that maintainers would prefer to be making a decision between this PR and #1195 up-front, rather than merging this PR and then contemplating even further revisions to the IP; especially if the validator needs to be compatible with both for different BIDS versions. At some point I'd like to make a draft PR for #1195 using the code of this PR as the base, which would highlight the specific change being proposed in #1195.

I'm also very interested to hear from anyone responsible for the validator or popular BIDS APIs, since the proposed changes are *highly* consequential there.

Reply:
well, I will then let you decide first which way to proceed (undraft this PR, or create one with #1195), since in principle I then should not even comment on "Draft" right, and there is only fractional value return from discussing something which has no potential on being merged ATM ;)

FWIW -- I consider this PR already a substantial improvement and #1195 resolution is not necessarily needed to proceed forward with it.

And FWIW - awaiting for feedback from other developers, in particular for a PR in Draft mode, might take another year.